fe/fe-core/src/main/java/org/apache/doris/job/extensions/insert/InsertJob.java

```diff
@@ -65,58 +100,171 @@ public class InsertJob extends AbstractJob<InsertTask, Map> {
+    // 0: the job status is pending
+    // n/100: n is the number of task which has been finished
+    // 99: all tasks have been finished
+    // 100: txn status is visible and load has been finished
+    @SerializedName("progress")
+    private int progress;
+    @SerializedName("failMsg")
+    private FailMsg failMsg;
+    @SerializedName("plans")
+    private List<InsertIntoTableCommand> plans;
+    private LoadStatistic loadStatistic = new LoadStatistic();
+    private Set<Long> finishedTaskIds = new HashSet<>();
+    private Set<String> tableNames;
+    private ConcurrentHashMap<Long, InsertTask> idToTasks = new ConcurrentHashMap<>();
+    private Map<String, String> properties;
+    private AuthorizationInfo authorizationInfo;
+    private final ReentrantReadWriteLock lock = new ReentrantReadWriteLock(true);
+
+    private ConnectContext ctx;
+    private StmtExecutor stmtExecutor;
+    private List<ErrorTabletInfo> errorTabletInfos = new ArrayList<>();
+    private List<TabletCommitInfo> commitInfos = new ArrayList<>();
+
+    // max save task num, do we need to config it?
+    private static final int MAX_SAVE_TASK_NUM = 100;
+
+    /**
+     * load job type
+     */
+    public enum LoadType {
+        BULK,
+        SPARK,
+        LOCAL_FILE,
+        UNKNOWN
+
+    }
+
+    public enum Priority {
+        HIGH(0),
+        NORMAL(1),
+        LOW(2);
 
-    InsertIntoTableCommand command;
+        Priority(int value) {
+            this.value = value;
+        }
 
-    StmtExecutor stmtExecutor;
+        private final int value;
 
-    ConnectContext ctx;
+        public int getValue() {
+            return value;
+        }
+    }
 
-    @SerializedName("tis")
-    ConcurrentLinkedQueue<Long> taskIdList;
+    public InsertJob(Long jobId, String jobName,
+                     JobStatus jobStatus,
+                     LabelName labelName,
+                     String comment,
+                     UserIdentity createUser,
+                     JobExecutionConfiguration jobConfig,
+                     Long createTimeMs,
+                     String executeSql) {
+        super(jobId, jobName, jobStatus, labelName.getDbName(), comment, createUser,
+                jobConfig, createTimeMs, executeSql, null);
+        this.dbId = ConnectContext.get().getCurrentDbId();
+        this.labelName = labelName.getLabelName();
+    }
 
-    // max save task num, do we need to config it?
-    private static final int MAX_SAVE_TASK_NUM = 100;
+    public InsertJob(ConnectContext ctx,
+                      StmtExecutor executor,
+                      String labelName,
+                      List<InsertIntoTableCommand> plans,
+                      Set<String> sinkTableNames,
+                      Map<String, String> properties,
+                      String comment,
+                      JobExecutionConfiguration jobConfig) {
+        super(Env.getCurrentEnv().getNextId(), labelName, JobStatus.RUNNING, null,
```

Comment:
The generation of job id be both generated outside or inside.
In first constructor, it is passed from outside, in second constructor, is is generated inside.
We should unify them